### PR TITLE
dependencies: Bump riscemu to 2.2.5

### PR DESCRIPTION
**Pull request by AntonLydike**
Beep Boop, I am dependabot. I want to bump riscemu to 2.2.5.

### CODE REVIEW

**Comment by AntonLydike** [comment]
narghhhh




I'll get to it tomorrow

**Comment by math-fehr** [comment]
Good bot

**Comment by AntonLydike** [comment]
Okay, I fixed riscemu, just need to wait a bit for caches to clear, apparently CI can't find the newly released version quite yet...

**Comment by AntonLydike** [comment]
@superlopuh I think I need help with this filecheck here: filecheck/with-riscemu/riscv_emulation.mlir

We now need every module to declare a global `main` symbol, otherwise riscemu won't pick it up (when running with `-o libc`, which is preferable.

**Comment by AntonLydike** [comment]
I guess the `.globl main` directive is not added, maybe a lowering is missed? Should I add it manually?

File: tests/interpreters/test_riscv_emulator.py [hunk 1]
```diff
@@ -30,22 +30,24 @@ def body():
...
             )
+            riscv.ReturnOp()
 
         riscv_func.FuncOp("main", body, ((), ()))
 
     RISCVRegisterAllocation().apply(ctx, module)
 
     code = riscv.riscv_code(module)
 
+    # TODO: fix the adding of .text .globl main
```
Comment (superlopuh) [inline tests/interpreters/test_riscv_emulator.py:41]: Can you please add a `riscv.DirectiveOp(".globl", ".main")`? Should be fairly quick

File: tests/interpreters/test_riscv_emulator.py [hunk 2]
```diff
@@ -30,22 +30,24 @@ def body():
...
     RV_Debug.stream = stream
     run_riscv(
-        code,
+        ".text\n.globl main\n" + code,
         extensions=[RV_Debug],
         unlimited_regs=True,
         verbosity=1,
     )
-    assert "42\n" == stream.getvalue()
+    assert stream.getvalue() == "42\n"
```
Comment (superlopuh) [inline tests/interpreters/test_riscv_emulator.py:50]:
```suggestion
    assert stream.getvalue() == "42\n"
```
:pray:

**Review by superlopuh — APPROVED** [review]
TYVM, just one minor thing to change, and we're GTG. Great to see my hacks being removed

File: tests/interpreters/test_riscv_emulator.py (hunk 2 above)
Comment (AntonLydike) [reply tests/interpreters/test_riscv_emulator.py:51]: Not sure what you are suggesting here?

**Review by math-fehr — APPROVED** [review]
I see the numbers of lines go down, I press approve.

File: tests/interpreters/test_riscv_emulator.py (hunk 2 above)
Comment (superlopuh) [reply tests/interpreters/test_riscv_emulator.py:51]: No I am thanking you :)

**Pull request merged**